### ISSUE TITLE
Pathfinding. Ships going to Europe choose paths longer than necessary.

### ISSUE THREAD

**Issue by windock** [We-the-People-civ4col-mod/Mod#610]
Develop 2 branch.

All the ships going from my main port city choose path to Europe (Sail to Europe menu item) that is longer than necessary. 

On the first screenshot my galleon is going to Europe. I have, for demonstration purposes, plotted it's possible path, by right-clicking on the border of the New World. As you can see, it needs 1 turn only to go to the edge, with a single square extra of pathing available.
![turn1](https://user-images.githubusercontent.com/79237/184428743-58581a4a-3a24-410e-895c-ddce39cd0bdd.png)

This screenshot is after I end the current turn. You can see that the ship, instead of going East, went South-East. It lost a full turn by that maneuver, comparing to the path of the previous turn.
![turn2](https://user-images.githubusercontent.com/79237/184428756-9f757487-5cea-4a09-ba79-9fa4d9b64165.png)

Attaching two save files, corresponding to 2 turns in screenshots.
[windock AD-1660-July.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9329407/windock.AD-1660-July.zip)
[windock AD-1661-January.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9329414/windock.AD-1661-January.zip)

**Comment by MrZorG33** [comment]
this statement is correct and about the way from Europe to the colonies.
and where is the point of the first appearance of the ship on the first turn? I have an assumption that the ship is sailing towards it. I may be wrong)

**Comment by windock** [comment]
@MrZorG33 
The first appearance of the ship (on first screenshot) is needed to show that the ship can go to Europe on the next turn. It shows a path that I would use to manually move to Europe.
However, if you order it to move to Europe instead automatically, next turn is shown on next screenshot.
Second turn’s screenshot shows that the same ship, while ordered to go to Europe, chooses a path that wastes 1 full turn of movement.

**Comment by windock** [comment]
in the following turns of the game ships sometimes choose a better path, going strait East. And they get back from Europe using the same efficient path. Maybe the issue is ships try to avoid any terrain features (unfavorable wind, favorable wind).

**Comment by raystuttgart** [comment]
Also possible, since they are actually considered in PathFinding.
(e.g. Damage that the Ship could suffer.)

**Comment by raystuttgart** [comment]
Yeah I think this is really just related to Terrain Features like Winds and Storms.
Because those can also hurt the Ships. I think the logic is fine.

**Comment by MrZorG33** [comment]
I'll put in my 50 cents
according to my observations, ships going to Europe try to move towards the approximate tile of their first appearance. I noticed this in my last test game. I think this needs to be rechecked. because the path increased by about three times.

**Comment by MrZorG33** [comment]
one more addition.
my karakka was shipped to europe in auto mode. when, according to my calculations, she should have already been on the docks of Europe, I found her on the map at this place.

we need to figure out why the ships ignore the first tiles "access Europe". and to what point are they moving?))
![image](https://user-images.githubusercontent.com/61798883/223706735-d4eb17a3-10c2-4895-b1ec-66b1d1412c31.png)

**Comment by devolution79** [comment]
Upload a save or it didn't happen :P

**Comment by raystuttgart** [comment]
Also as I wrote in the forum, please check if the **correct Europe Plots** were placed. 👍 


That might also cause Ships to take additional moves before they sail to Europe.
(It is actually pretty easy to check and correct.)

**Comment by devolution79** [comment]
Remember that the AI (and automated units by the player) base their decision on what they can currently see. That means that due to the weather effects or hostile units (pirates, animals) they may choose a path that appears sub-optimal.
As for the current, it should be ignored for this discussion since the pathfinder has no support for it and this it's not a bug but a missing feature!

As for the ship ignoring the nearest Europe plots, I'd need a save or a instructions on how to reproduce. It would also be interesting to know if it applies to random maps as well.

**Comment by devolution79** [comment]
1)
It's a bit hard to tell since the pictures are not using the grid, which I greatly prefer. Note that the pathfinder does not currently support the +1 movement that can be gained from certain water features (favorable winds?) so it will not understand that pathing through such features could speed up movement.

2)
The picture further above, that shows a ship failing to sail to Europe when on an actual Europe plot, is a clear bug though.

**Comment by MrZorG33** [comment]
> Note that the pathfinder does not currently support the +1 movement that can be gained from certain water features (favorable winds?) so it will not understand that pathing through such features could speed up movement.

I can argue the opposite, since when receiving the appropriate promotion, automated ships use such tiles to speed up their movement.

**Comment by MrZorG33** [comment]
the problem is still relevant. 
On this screenshot you can see the position of the Schooner before I pressed the "sail to Europe" button.
![Image](https://github.com/user-attachments/assets/4943c44c-c4e7-4a2b-9da7-85fce4268a40)

On this screenshot you can see that instead of the shortest route, the schooner chose a route several times longer.
![Image](https://github.com/user-attachments/assets/0f7c09be-38a4-42be-ba10-481d1f9b3bdc)

On the next turn the schooner headed to Europe from the tile on which it finished the previous turn.

[dynamic_test.zip](https://github.com/user-attachments/files/19314484/dynamic_test.zip)

**Comment by devolution79** [comment]
Ok, I see the problem. We are attempting to minimize turns and in this case the schooner does reach the Europe plot this turn. However, since it has no movement points left it cannot sail to Europe. The solution is probably add the remaining moves as a tie-breaker so that we prefer direct paths (and this would also preserve enough movement points to also enter Europe)

**Comment by MrZorG33** [comment]
~~Houston~~, @devolution79  we have a problem
load the save, open the Europe window and send the Schooner to Santos.
![Image](https://github.com/user-attachments/assets/4dd87715-e75b-4ff8-ba80-bd30846a76c2)

сlose the Europa window and end your turn. аfter the start of the next turn during the pirate event, do not pay them, open the Europa window and see where the ship went. for me it sailed to Rio de Janeiro.

![Image](https://github.com/user-attachments/assets/6e625ff7-83e0-4c51-9e89-4ea2a9ad65de)

in English the destination path is not shown to me for some reason.
![Image](https://github.com/user-attachments/assets/50d65e1c-747b-495e-b583-3291d225e4dd)

[test.zip](https://github.com/user-attachments/files/19317094/test.zip)

**update**: if you pay the pirates, the destination will not change.

**Comment by devolution79** [comment]
That seem like a completely separate issue. We need a separate issue for it IMHO

**Comment by MrZorG33** [comment]
@devolution79 ok, i'll create a new one.
